Sources/StreamChat/APIClient/Endpoints/Payloads/UserPayloads.swift

```diff
@@ -40,6 +40,27 @@ class UserPayload<ExtraData: UserExtraData>: Decodable {
     let teams: [String]
     let extraData: ExtraData
     
+    static func == (lhs: UserPayload<ExtraData>, rhs: UserPayload<ExtraData>) -> Bool {
+        lhs.id == rhs.id
+    }
+    
+    func hash(into hasher: inout Hasher) {
+        hasher.combine(id)
+        hasher.combine(name)
+        hasher.combine(imageURL)
+        hasher.combine(role.rawValue)
+        hasher.combine(createdAt)
+        hasher.combine(updatedAt)
+        hasher.combine(lastActiveAt)
+        hasher.combine(isOnline)
+        // hasher.combine(isInvisible)
+        hasher.combine(isBanned)
+        // hasher.combine(teams)
+        if let extraData = try? JSONEncoder.default.encode(extraData) {
+            hasher.combine(extraData)
+        }
+    }
+    
```

Comment:
There are some big statements here. Implementing `==` just as `id == id` in terms of `UserPayload` is IMO not correct. This is a payload describing an entity, not the entity itself (where `id == id` makes more sense).

I think we shouldn't conform to hashable explicitly unless it's a _true_ hash -> like the one generated by the compiler.

Isn't all we need just some kind of method like `computeHashForDTOSerializing()`?

Reply:
Very good fundemantel remarks here. I also came to the realization that `Hashable` is actually not correct in this case. We're using hashes, but for our own purpose. I think we need to have our own protocol, so both DTO and payload and implement it.